lxd/device/disk.go

```diff
@@ -118,7 +118,19 @@ func (d *disk) sourceIsCeph() bool {
 
 // CanHotPlug returns whether the device can be managed whilst the instance is running.
 func (d *disk) CanHotPlug() bool {
-	return !(d.sourceIsDir() || d.sourceIsCephFs()) || d.inst.Type() == instancetype.Container
+	// Containers support hot-plugging all disk types.
```

Comment:
`lxc config device add v1 d1 disk source=vol1 pool=default`

Reply:
We shouldnt allow adding dir based or cephfs shares to a VM (or container for that matter) without a target `path` so that is the more general approach to indicate a filesystem disk.

Equally we shouldn't allow a block disk to have a `path` for VMs.

Reply:
The command you provided already returns `Error: Failed add validation for device "d1": Custom filesystem volumes require a path to be defined`. Am I missing something?

But anyway I might have tested only with directories on the host, not custom volumes of type filesystem... 😇

Reply:
Ill get a reproducer...moved on to other things now...

Reply:
Thanks!

Reply:
Oh I think it was possible to attach without a "path" defined, which then caused a qemu error as we were trying to attach a filesystem as a block device, and also it was possible to attach a filesystem volume with a path, start the VM, and then it allowed live unattach which caused that hang again.

Reply:
From main:

```
lxc storage volume create default vol1fs --type=filesystem
lxc storage volume create default vol1block --type=block
lxc launch ubuntu:jammy v1 --vm

lxc config device add v1 vol1 disk source=vol1fs pool=default 
Error: Failed add validation for device "vol1": Custom filesystem volumes require a path to be defined

# Bug 1
lxc config device add v1 vol1 disk source=vol1fs pool=default path=/mnt
Error: Failed to start device "vol1": Failed to add drive config: Invalid device path format "36"

lxc stop v1
lxc config device add v1 vol1 disk source=vol1fs pool=default path=/mnt
lxc start v1

# Bug 2
lxc config device remove v1 vol1
Device vol1 removed from v1
lxc exec v1 -- ls /mnt #hangs
```

Reply:
With branch:

```
lxc launch ubuntu:jammy v1 --vm

lxc config device add v1 vol1 disk source=vol1fs pool=default 
Error: Failed add validation for device "vol1": Custom filesystem volumes require a path to be defined

lxc config device add v1 vol1 disk source=vol1fs pool=default path=/mnt
Error: Failed to add device "vol1": Device cannot be added when instance is running

lxc stop v1

lxc config device add v1 vol1 disk source=vol1fs pool=default path=/mnt
lxc start v1
lxc config device remove v1 vol1
Error: Failed to stop device "vol1": Device cannot be stopped when instance is running
```
```

Reply:
Where is the double approve button :) Thx for the follow up.